models/app.models.js

```diff
@@ -1,43 +1,145 @@
 const db = require('../db/connection');
-const { convertTimestampToDate } = require('../db/helpers/utils')
+const {
+  convertTimestampToDate,
+  formatComments,
+} = require('../db/helpers/utils');
 
-exports.getAllTopics = async () => {
-    const results = await db.query('SELECT * FROM topics;');
-        return results.rows;
-    };
-
-exports.getAllUsernames = async () => {
-    const results = await db.query('SELECT username FROM users');
-        return results.rows;
+exports.fetchAllTopics = async () => {
+  const results = await db.query('SELECT * FROM topics;');
+  return results.rows;
 };
 
-exports.getArticleFromDbById = async (articleId) => {
-            const results = await db.query(`SELECT * FROM articles WHERE article_id=$1`, [articleId]);
-            if (results.rows.length) {
-                //Covert timestamp to date using helper.
-                const convertedResults = convertTimestampToDate(results.rows);
-                return convertedResults["0"];
-            } else {
-                return Promise.reject({msg: 'Not Found', status: 404});  
-            }
+exports.fetchAllUsernames = async () => {
+  const results = await db.query('SELECT username FROM users');
+  return results.rows;
 };
 
-exports.getAllArticles = async () => {
+exports.fetchArticleFromDbById = async (articleId) => {
   const results = await db.query(
-    `SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, COUNT(articles.article_id)
-    AS comment_count 
-    FROM articles 
-    FULL OUTER JOIN comments 
-    ON articles.article_id = comments.article_id 
-    GROUP BY articles.article_id
-    ORDER BY articles.created_at DESC;`
+    `SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.body, articles.votes, COUNT(articles.article_id)
+            AS comment_count 
+            FROM articles 
+            FULL OUTER JOIN comments 
+            ON articles.article_id = comments.article_id 
+            WHERE articles.article_id=$1
+            GROUP BY articles.article_id
+            ORDER BY articles.created_at DESC`,
+    [articleId]
   );
+  if (results.rows.length) {
+    //Covert timestamp to date using helper.
+    const convertedResults = convertTimestampToDate(results.rows);
+    return convertedResults['0'];
+  } else {
+    return Promise.reject({ msg: 'Not Found', status: 404 });
+  }
+};
+
+exports.fetchAllArticles = async (
+  sort_by = 'created_at',
+  order = 'DESC',
+  topic
+) => {
+  let query = `SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.body, articles.votes, COUNT(articles.article_id)
+  AS comment_count 
+  FROM articles 
+  FULL OUTER JOIN comments 
+  ON articles.article_id = comments.article_id`;
+
+  const validTopics = ['mitch', 'cats'];
```

Comment:
Which topics are valid depends on what's in the topics table. You'll need to query the db to find this out and can't hard code these values as they're specific to your test data.